### PR TITLE
Add connectivity check queue

### PR DESCRIPTION
**Pull request by gregor**

### CODE REVIEW

File: test/unit_tests/service/ClientSpec.coffee [hunk 1]
```diff
@@ -52,6 +52,7 @@ describe 'z.service.Client', ->
         expect(client.status).toHaveBeenCalledTimes 1
         done()
       .catch done.fail
+      jasmine.clock().tick 10
```
Comment (bennycode) [inline test/unit_tests/service/ClientSpec.coffee:55]: `10` is a pretty short value. Does it need to be that short or is `100` also ok?

File: app/script/service/Client.coffee [hunk 2]
```diff
@@ -92,30 +108,40 @@ class z.service.Client
...
+  execute_on_connectivity: (source) =>
+    source = z.service.Client::CONNECTIVITY_CHECK_TRIGGER.UNKNOWN if not source
+    @logger.info "Connectivity check requested by '#{source}'"
+
+    _check_status = =>
+      @status()
+      .done (jqXHR) =>
+        @logger.info 'Connectivity verified', jqXHR
+        @connectivity_timeout = undefined
+        @connectivity_queue.pause(false)
```
Comment (bennycode) [inline app/script/service/Client.coffee]: In CofeeScript land it is `@connectivity_queue.pause false` 😁

File: app/script/service/Client.coffee [hunk 3]
```diff
@@ -92,30 +108,40 @@ class z.service.Client
...
+      .fail (jqXHR) =>
+        if jqXHR.readyState is 4
+          @logger.info "Connectivity verified by server error '#{jqXHR.status}'", jqXHR
+          @connectivity_queue.pause(false)
```
Comment (bennycode) [inline app/script/service/Client.coffee]: Parentheses not needed in CoffeeScript

File: app/script/service/Client.coffee [hunk 4]
```diff
@@ -92,30 +108,40 @@ class z.service.Client
...
+          @connectivity_timeout = undefined
+        else
+          @logger.warn 'Connectivity could not be verified... retrying'
+          @connectivity_queue.pause(true)
```
Comment (bennycode) [inline app/script/service/Client.coffee]: Parentheses not needed in CoffeeScript

File: test/unit_tests/service/ClientSpec.coffee (hunk 1 above)
Comment (gregor) [reply test/unit_tests/service/ClientSpec.coffee:55]: It just needs to be geater than 0 as I added a setTimeout to the initial call

**Pull request merged**